modules/howtos/pages/encrypting-using-sdk.adoc

```diff
@@ -1,15 +1,173 @@
 
 [abstract]
-{description}
+{description}
 
+For a high-level overview of this feature, see our xref:concept-docs:encryption.adoc[Encryption] page.
 
-Client-side implementation of Field Level Encryption is available in the previous version of the SDK.
-It will be enabled in a future release of the third generation SDK.
+[#package]
+== Packaging
+The Node.js SDK works together with the https://github.com/couchbase/node-couchbase-encryption[Node Couchbase Encryption^] library to provide support for encryption and decryption of JSON fields.
+This library includes cryptographic algorithms you can use out of the box, and provides a framework for implementing your own crypto components.
 
-A _Developer Preview_ is available in the xref:3.0@java-sdk:howtos:encrypting-using-sdk.adoc[Java SDK].
+NOTE: The encryption code is packaged as an optional library because it is released under a https://github.com/couchbase/node-couchbase-encryption/blob/master/LICENSE[different license^] from the Couchbase Node SDK.
+To use the encryption library, you have to explicitly include this dependency in your project configuration.
+
+To get started with the Node encryption library you can fetch it using:
+
+[source,console]
+----
+$ npm i cbfieldcrypt
+----
+
+== Configuration
+The Node.js Field-Level Encryption library works on the principle of `Encrypters` and `Decrypters` which can be packaged within a `Provider`, as well as a custom xref:transcoders-nonjson.adoc[`Transcoder`].
+`Encrypters` and `Decrypters` are registered with a `CryptoManager` and are then used at serialization/deserialization time to encrypt and decrypt fields.
+
+Here we'll go through an example of setting up and using the Node Field-Level Encryption library.
+
+To begin we need to create a couple of keys, you should *not* use the `InsecureKeyring` other than for evaluation purposes and should keep your keys secure.
+
+[source,javascript,indent=0]
+----
+include::example$fle.js[tag=keys]
+----
+
+Now that we have keys we can create a `Provider` (here we use the `AeadAes256CbcHmacSha512` algorithm which is the default supplied by the library).
+The `Provider` gives us a way to easily create multiple encrypters for the same algorithm but different keys.
+At this point we also create `CryptoManager` and register our encrypters and decrypters with it.
+
+[source,javascript,indent=0]
+----
+include::howtos:example$fle.js[tag=provider]
+----
+
+== Usage
+
+Sensitive fields in your data classes can be encrypted by using a `CryptoSchema`. For example:
+
+[source,javascript,indent=0]
+----
+include::howtos:example$fle.js[tag=crypto-schema]
+----
+
+Now let's create a person document and save it to Couchbase:
+
+[source,javascript,indent=0]
+----
+include::howtos:example$fle.js[tag=upsert]
+----
+
+You can get the document to verify the fields were encrypted:
+
+[source,javascript,indent=0]
+----
+include::howtos:example$fle.js[tag=get]
+----
+
+The expected output is something like:
+
+[source,console]

```

Comment:
could be json in this case

Reply:
hmm, I think the other [output](https://github.com/couchbase/docs-sdk-nodejs/pull/118/files/91c24d89d6dbcf89d58eb28fb73e71e8803860b2#diff-9f05559f0fa7c210b5a3b6e9db62a809a637c7af0af1e5701b10b3d8b90b6623R106-R128) could be json as well, will update 👍